File: VB/DocumentSelectorApp/ChildForm.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports DevExpress.Utils
Imports DevExpress.XtraEditors
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Views.Base

Namespace DevExpress.Samples.DocumentSelector
	Partial Public Class ChildForm
		Inherits XtraForm
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Shared groupIndex As Integer = 0
		Private Sub Form2_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			AddHandler gridView1.FocusedRowChanged, AddressOf Form2_FocusedRowChanged
			Dim photos As IList(Of Photo) = New List(Of Photo)()
			Dim group() As Integer = photoGroups((groupIndex) Mod photoGroups.Length)
			groupIndex += 1
			For i As Integer = 0 To group.Length - 1
				photos.Add(New Photo(String.Format("Photo{0}", group(i)), String.Format("Images\0{0}.jpg", group(i))))
			Next i
			gridControl1.DataSource = photos
			gridView1.BestFitColumns()
		End Sub
		Private Sub Form2_FocusedRowChanged(ByVal sender As Object, ByVal e As FocusedRowChangedEventArgs)
			If e.FocusedRowHandle = GridControl.InvalidRowHandle Then
				Return
			End If
			pictureBox1.Image = LoadImage(CStr(gridView1.GetRowCellValue(e.FocusedRowHandle, "Path")))
		End Sub
		Private Shared Function LoadImage(ByVal path As String) As System.Drawing.Image
			Dim currentAssembly As System.Reflection.Assembly = System.Reflection.Assembly.GetAssembly(GetType(ChildForm))
			Return ResourceImageHelper.CreateImageFromResources("DevExpress.Samples.DocumentSelector." & path.Replace("\", "."), currentAssembly)
		End Function
		Private Shared photoGroups()() As Integer = { New Integer() { 5, 3, 7 }, New Integer() { 3, 4 }, New Integer() { 2, 3, 4, 5 }, New Integer() { 0, 1, 6, 8 } }
	End Class
	Friend Class Photo
		Public Sub New(ByVal name As String, ByVal path As String)
			nameCore = name
			pathCore = path
		End Sub
		Private nameCore As String
		Private pathCore As String
		Public ReadOnly Property Name() As String
			Get
				Return nameCore
			End Get
		End Property
		Public ReadOnly Property Path() As String
			Get
				Return pathCore
			End Get
		End Property
	End Class
End Namespace

File: CS/DocumentSelectorApp/ChildForm.cs
using System;
using System.Collections.Generic;
using DevExpress.Utils;
using DevExpress.XtraEditors;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Views.Base;

namespace DevExpress.Samples.DocumentSelector {
    public partial class ChildForm : XtraForm {
        public ChildForm() {
            InitializeComponent();
        }
        static int groupIndex = 0;
        void Form2_Load(object sender, EventArgs e) {
            gridView1.FocusedRowChanged += Form2_FocusedRowChanged;
            IList<Photo> photos = new List<Photo>();
            int[] group = photoGroups[(groupIndex++) % photoGroups.Length];
            for(int i = 0; i < group.Length; i++) {
                photos.Add(
                    new Photo(
                        string.Format("Photo{0}", group[i]),
                        string.Format(@"Images\0{0}.jpg", group[i])
                    ));
            }
            gridControl1.DataSource = photos;
            gridView1.BestFitColumns();
        }
        void Form2_FocusedRowChanged(object sender, FocusedRowChangedEventArgs e) {
            if(e.FocusedRowHandle == GridControl.InvalidRowHandle) return;
            pictureBox1.Image = LoadImage((string)gridView1.GetRowCellValue(e.FocusedRowHandle, "Path"));
        }
        static System.Drawing.Image LoadImage(string path) {
            System.Reflection.Assembly currentAssembly = System.Reflection.Assembly.GetAssembly(typeof(ChildForm));
            return ResourceImageHelper.CreateImageFromResources(
                "DevExpress.Samples.DocumentSelector." + path.Replace(@"\", "."), currentAssembly);
        }
        static int[][] photoGroups = new int[][]{
            new int[] { 5, 3, 7 },
            new int[] { 3, 4 },
            new int[] { 2, 3, 4, 5 },
            new int[] { 0, 1, 6, 8 }
        };
    }
    class Photo {
        public Photo(string name, string path) {
            nameCore = name;
            pathCore = path;
        }
        string nameCore;
        string pathCore;
        public string Name {
            get { return nameCore; }
        }
        public string Path {
            get { return pathCore; }
        }
    }
}